enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -0,0 +1,379 @@
+  from https://github.com/openshift/aws-ebs-csi-driver-operator repository.
+
+* During OCP 4.15 development, we want ART to build both `ose-aws-ebs-csi-driver-operator-container` (as today)
+  and say `ose-aws-ebs-csi-driver-operator-v2-container` from https://github.com/openshift/csi-operator.
+  TODO: better name for the new images?
+
+  We will file these tickets / PRs ([source](https://art-dash.engineering.redhat.com/self-service/new-content)):
+  * Comet:
+    * New _build repository_ for `ose-aws-ebs-csi-driver-operator-v2-container`.
+    * New _delivery repository_ for `ose-aws-ebs-csi-driver-operator-v2-container`.
+    * Shall we follow the same brew / dist-git / image names as the old one, just add `-v2-`? Or shall we fix the inconsistencies (e.g. `azure-file-...` vs `ose-azure-disk-...`)? 
+  * openshift/release:
+    * build the image in CI + promote to 4.15.
+  * ART:
+    * Follow https://art-dash.engineering.redhat.com/self-service/new-content, using almost the same data as in the existing image (e.g. multiarch support)
+      * Exception: we will not perform threat model assessment - we're merging code from existing repos.
+    
+* Before OCP 4.15 feature freeze we will decide if the `-v2-` image is good enough and file tickets to
+  disable builds of either the old or `-v2-` image. Similarly, we will update `cluster-storage-operator` and its feature gate
+  to use only one of these images.
+  We will file these tickets + PRs ([source](https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#removing-a-component-from-the-openshift-release-payload)):
+  * PR to update `cluster-storage-operator` and its related-images, so only the "good" image is used.
+    * This can be merged before anything below.
+  * Jira for ART to stop building the "bad" image, https://issues.redhat.com/browse/ART-1443
+    * They will create a PR to update ocp-build-data.
+  * When removing the old image: PR to openshift/release to stop building + promoting the image in CI.
+  * On slack, coordinate with ART merging of all these PRs
+  
+* At 4.15 GA (and in the whole 4.15.z stream), ART will build and ship only the "good" image.
+
+* In 4.16, if the "good" image is the old one, we re-submit all tickets to enable building + shipping of the `-v2-` image
+  again and continue testing.
+    * Drawback: we will have to stop its development + testing until OCP 4.16 is branched, so we can keep 4.15 with only
+      a single image until GA.
+
+Drawbacks:
+
+* New images need to be built, i.e. lot of Comet requuests.
+* More tickets.
+
+Advantages:
+
+* Can be tested easily in CI and by QE.
+
+##### Option 2: Build & test images manually and do a hard switch
+
+During development of an OCP release, we will merge an CSI driver operator to github.com/openshift/csi-operator
+without any testing in CI. We will keep as close to the original operator as possible and we will test the new image
+manually.
```

Comment:
What component replaces the images in the [CSO manifest](https://github.com/openshift/cluster-storage-operator/blob/18bb6aa272ec729f6c7ce94a4dfff12d5f211f8e/manifests/10_deployment.yaml#L65-L138) with the release payload image?

Could we add some e2e jobs in openshift/release for the openshift/csi-operator repo that does pre-merge testing by replacing the image in the CSO deployment with the one built from openshift/csi-operator?

OLM operators like [EFS for example](https://github.com/openshift/release/blob/7a2c4562e5d0823b6313d59d563e5b5794584d84/ci-operator/config/openshift/aws-efs-csi-driver-operator/openshift-aws-efs-csi-driver-operator-master.yaml#L63-L75) have a way to replace the images in the CSV. I think we could find a way to do the same level of pre-merge testing with the operator built from openshift/csi-operator repo before changing anything in ART.

Reply:
I added a bullet point to the example below, I am not sure it would work.